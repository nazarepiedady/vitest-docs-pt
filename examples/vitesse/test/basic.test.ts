import { mount } from '@vue/test-utils'
import Hello from '../src/components/Hello.vue'

test('mount component', async () => {
  expect(Hello).toBeTruthy()

  const wrapper = mount(Hello, {
    props: {
      count: 4,
    },
  })

  expect(wrapper.text()).toContain('4 x 2 = 8')
  expect(wrapper.html()).toMatchSnapshot()

  await wrapper.get('button').trigger('click')
  await nextTick()

  expect(wrapper.text()).toContain('4 x 3 = 12')

  await wrapper.get('button').trigger('click')
  await wrapper.get('button').trigger('click')

  expect(wrapper.text()).toContain('4 x 5 = 20')
  expect(wrapper.html()).toMatchSnapshot()
})
