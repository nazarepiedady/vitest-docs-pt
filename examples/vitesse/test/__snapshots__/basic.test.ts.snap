// Vitest Snapshot v1

exports[`mount component 1`] = `
"<div class=\\"foo\\" data-v-958b189a=\\"\\">
  <div>4 x 2 = 8</div><button> x1 </button>
</div>"
`;

exports[`mount component 2`] = `
"<div class=\\"foo\\" data-v-958b189a=\\"\\">
  <div>4 x 5 = 20</div><button> x1 </button>
</div>"
`;
